clippy_lints/src/methods/iter_on_single_or_empty_collections.rs

```diff
@@ -43,6 +65,25 @@ pub(super) fn check(cx: &LateContext<'_>, expr: &Expr<'_>, method_name: &str, re
     let is_unified = match get_expr_use_or_unification_node(cx.tcx, expr) {
         Some((Node::Expr(parent), child_id)) => match parent.kind {
             ExprKind::If(e, _, _) | ExprKind::Match(e, _, _) if e.hir_id == child_id => false,
+            ExprKind::Call(
+                Expr {
+                    kind: ExprKind::Path(path),
+                    hir_id,
+                    ..
+                },
+                args,
+            ) => is_arg_ty_unified_in_fn(
+                cx,
+                cx.typeck_results().qpath_res(path, *hir_id).def_id(),
```

Comment:
The call to `def_id` here will ICE when calling a local variable as they don't have a `DefId`
```rs
let c = |_: std::slice::Iter<()>| {};
c([].iter());
```